syntax = "proto3";
package m2g.pb;

message ServerCommand {}

message ClientCommand {
  message Action {
    message BuildAction {}
    message NetworkAction {}
    message SellAction {}
    message LoanAction {}
    message DevelopAction {}
    message ScoutAction {}

    oneof type {
      BuildAction build_action = 1;
      NetworkAction network_action = 2;
      SellAction sell_action = 3;
      LoanAction loan_action = 4;
      DevelopAction develop_action = 5;
      ScoutAction scout_action = 6;
    }
  }

  Action first_action = 1;
  Action second_action = 2;
}
